lib/tapioca/runtime/reflection.rb

```diff
@@ -168,6 +168,17 @@ def required_from_location
 
         required_location.absolute_path || ""
       end
+
+      sig { params(constant: Module).returns(T.nilable(String)) }
+      def constant_name_from_singleton_class(constant)
```

Comment:
I moved these two methods into reflection because I had to use them in `Trackers::Mixin` to find the attached class based on a singleton class.